### PR TITLE
Random color

### LINKED ISSUE #111: Assign random color upon event and era creation

### PR DESCRIPTION
**Pull request by dhalverson**
### Code Highlights:
- Renames date_helper to helper_methods
- Updates create event and create era to pick random hex color if blank
- Addes mutation testing
- Adds helper methods testing

### Where should the reviewer start?
 Spec > Requests > GraphQL > Mutations > create_era_spec & create_event_spec
 Spec > Requests > GraphQL > helper_methods_spec
 App > GraphQL > Mutations > create_era & create_event
 APp > GraphQL > Mutations > helper_methods

### Have Tests Been Added?
- [ ] No.
- [ ] Yes, but all tests are not passing.
- [x] Yes, and all are passing.
### Issues:
* Addresses #
* Closes #111 
### Tracking Consistency:
- [x] added appropriate labels
- [x] My code follows the code style of this project and has removed all unnecessary annotations
- [x] I have added comments on my pull request, particularly in hard-to-understand areas
- [x] looked at PR preview to check spelling, syntax, formatting, and completion
- [x] Rubocop Violations:

### CODE REVIEW

**Review by AngelaGuardia — APPROVED** [review]
Nice teamwork on this one all!

File: app/graphql/mutations/eras/create_era.rb [hunk 1]
```diff
@@ -9,14 +9,15 @@ class Mutations::Eras::CreateEra < ::Mutations::BaseMutation
...
 
   def resolve(attributes)
     start_date = attributes[:start_date].to_date
     end_date = attributes[:end_date].to_date
     user_bday = User.where(id: attributes[:user_id])[0].birthdate
-    attributes[:start_week] = Mutations::DateCalculator.week_number(start_date, user_bday)
-    attributes[:end_week] = Mutations::DateCalculator.week_number(end_date, user_bday)
+    attributes[:start_week] = Mutations::HelperMethods.week_number(start_date, user_bday)
+    attributes[:end_week] = Mutations::HelperMethods.week_number(end_date, user_bday)
+    attributes[:color] = Mutations::HelperMethods.set_color if attributes[:color].empty?
```
Comment (LHJE) [inline app/graphql/mutations/eras/create_era.rb:20]: Terrific!

File: app/graphql/mutations/eras/update_era.rb [hunk 2]
```diff
@@ -11,15 +11,15 @@ class Mutations::Eras::UpdateEra < ::Mutations::BaseMutation
...
   def resolve(attributes)
     era = Era.find(attributes[:id])
     start_date = attributes[:start_date].to_date
     end_date = attributes[:end_date].to_date
     user_bday = User.where(id: era[:user_id])[0].birthdate
-    attributes[:start_week] = Mutations::DateCalculator.week_number(start_date, user_bday)
-    attributes[:end_week] = Mutations::DateCalculator.week_number(end_date, user_bday)
+    attributes[:start_week] = Mutations::HelperMethods.week_number(start_date, user_bday)
+    attributes[:end_week] = Mutations::HelperMethods.week_number(end_date, user_bday)
     era.update(attributes)
```
Comment (LHJE) [inline app/graphql/mutations/eras/update_era.rb:23]: Great Change

File: app/graphql/mutations/helper_methods.rb [hunk 3]
```diff
@@ -0,0 +1,10 @@
+module Mutations::HelperMethods
+  def self.week_number(date, user_bday)
+    (date - user_bday).to_i / 7
+  end
+
+  def self.set_color
+    ['#F7A83E', '#A94460', '#DB4709', '#96B40D', '#E7C408', '#70D6FF', '#FF70A6', '#AA9770', '#FFD670',
+     '#E9FF70'].sample
+  end
```
Comment (LHJE) [inline app/graphql/mutations/helper_methods.rb:9]: Love it!

**Review by LHJE — APPROVED** [review]
Great job!

**Pull request merged**